File: m3.py
import re
from urllib.parse import urlparse, parse_qs
def procurar_links_arquivo(nome_arquivo):
    # Padrão de expressão regular para procurar links
    padrao_link = r"http://[^\s'\"<>]+(?:=ts|plus)\b"

    # Abrir o arquivo para leitura
    with open(nome_arquivo, "r") as arquivo:
        conteudo = arquivo.read()

        # Procurar links usando a expressão regular
        links_encontrados = re.findall(padrao_link, conteudo)

        # Retornar os links encontrados
        return links_encontrados

# Exemplo de uso
nome_do_arquivo = "listas.txt"  # substitua pelo nome do arquivo que deseja procurar os links
links_encontrados = procurar_links_arquivo(nome_do_arquivo)
# Função para extrair o domínio, nome de usuário e senha de uma URL
def extrair_informacoes_link(url):
    parsed_url = urlparse(url)
    domain = parsed_url.netloc
    query_params = parse_qs(parsed_url.query)
    username = query_params.get('username', [''])[0]
    password = query_params.get('password', [''])[0]

    return domain, username, password

# Exemplo de uso
nome_do_arquivo = "listas.txt"  # substitua pelo nome do arquivo que deseja procurar os links
links_encontrados = procurar_links_arquivo(nome_do_arquivo)

# Processar cada link e extrair as informações desejadas
for link in links_encontrados:
    domain, username, password = extrair_informacoes_link(link)
    print("Link:", link)
    print("Domínio:", domain)
    print("Nome de usuário:", username)
    print("Senha:", password)
    print("\n\n")  # Pular linha entre os resultados

File: m3.js
const fs = require('fs');
const { URL } = require('url');
const axios = require('axios');
const iptv = require('iptv-playlist-parser');

function procurarLinksArquivo(nomeArquivo) {
  const conteudo = fs.readFileSync(nomeArquivo, 'utf8');
  const padraoLink = /http:\/\/[^\s'"<>]\b/g;
  const linksEncontrados = conteudo.match(padraoLink) || [];
  return linksEncontrados;
}

function extrairInformacoesLink(link) {
  const parsedUrl = new URL(link);
  const domain = parsedUrl.origin;
  const searchParams = parsedUrl.searchParams;
  const username = searchParams.get('username') || '';
  const password = searchParams.get('password') || '';
  return { domain, username, password };
}

async function verificarLinkIPTV(link) {
  try {
    const response = await axios.get(link, { validateStatus: false });
    const playlist = iptv.parse(response.data);
    if (playlist.items && playlist.items.length > 0) {
      return 'Funcionando';
    } else {
      return 'Playlist vazia ou inválida';
    }
  } catch (error) {
    return 'Erro de conexão';
  }
}

const nomeDoArquivo = 'tv.txt';
const linksEncontrados = procurarLinksArquivo(nomeDoArquivo);
(async () => {
  for (const link of linksEncontrados) {
    const { domain, username, password } = extrairInformacoesLink(link);
    console.log('Host:', domain);
    console.log('Nome de usuário:', username);
    console.log('Senha:', password);
    console.log('Link:', link);
    const status = await verificarLinkIPTV(link);
    console.log('Status:', status);

    console.log('\n');
  }

  console.log('Resultados:', linksEncontrados.length);
})();
